### PR TITLE
Adds new parameter for render method to allow custom variables for view script

### PR DESCRIPTION
**Pull request by froschdesign**
|    Q          |   A
|-------------- | ------
| Enhancement   | yes

### CODE REVIEW

File: src/Paginator.php [hunk 1]
```diff
@@ -791,18 +791,21 @@ public function normalizePageNumber($pageNumber)
...
-     * @param  \Laminas\View\Renderer\RendererInterface $view
+     * @param View\Renderer\RendererInterface|null $view
+     * @param array                                $params
      * @return string
      */
-    public function render(View\Renderer\RendererInterface $view = null)
-    {
+    public function render(
+        View\Renderer\RendererInterface $view = null,
+        array $params = []
```
Comment (michalbundyra) [inline src/Paginator.php:800]: @froschdesign adding argument to public method which is not final/internal (also class is not final/internal) is a BC Break. Added label and milestone for that PR.

**Review by michalbundyra — APPROVED** [review]
As noted in the comment below - it is BC Break, so added for 3.0.0 milestone.

File: src/Paginator.php (hunk 1 above)
Comment (froschdesign) [reply src/Paginator.php:800]: Correct! 👍 
This pull request can wait and was only opened so that I do not forget the feature myself.

Comment (weierophinney) [reply src/Paginator.php:800]: Um.. adding an _optional_ parameter is not a BC break UNLESS it's done in an interface.

Yes, those extending the class _could_ add more arguments in their extensions. However, that then becomes undefined behavior when it comes to the original class being extended.

As such, this is not a BC break, and can be added in a new minor release.

**Comment by froschdesign** [comment]
The rendering will be extracted in the next major version, so this extension is not needed.

**Pull request closed without merging**